st2common/st2common/metrics/base.py

```diff
@@ -24,11 +26,49 @@
 from st2common.util.date import get_datetime_utc_now
 from st2common.exceptions.plugins import PluginLoadError
 
+if not hasattr(cfg.CONF, 'metrics'):
+    from st2common.config import register_opts
+    register_opts()
+
+LOG = logging.getLogger(__name__)
 
 PLUGIN_NAMESPACE = 'st2common.metrics.driver'
 METRICS = None
 
 
+def _format_metrics_key_for_action_db(action_db):
+    action_name = action_db.name
+    action_pack = action_db.pack
+    return '.%s.%s' % (action_pack, action_name)
+
+
+def _format_metrics_key_for_liveaction_db(liveaction_db):
+    action_name = liveaction_db.action
+    action_pack = liveaction_db.context.get('pack', 'unknown')
+    return '.%s.%s' % (action_pack, action_name)
+
+
+def format_metrics_key(action_db=None, liveaction_db=None, key=None):
+    """Return a string for usage as metrics key.
+    """
+    assert (action_db or key or liveaction_db), """Must supply one of key, action_db, or
+                                                 liveaction_db"""
+    metrics_key = 'st2'
+
+    if action_db:
+        metrics_key += _format_metrics_key_for_action_db(action_db)
```

Comment:
One option would also be to return a string without a prefix from each of those utility functions and then join it using a separator (.), but current approach also works.